cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -525,6 +525,7 @@ public PartitionOutput getPartitionOutput(PartitionKey key) {
       throw new UnsupportedOperationException(
         "Output is not supported for external partitioned file set '" + spec.getName() + "'");
     }
+    validatePartitionKey(key, partitioning);
     return new BasicPartitionOutput(this, getOutputPath(key), key);
```

Comment:
Thanks. DynamicPartitioner directly uses it, so this protects that use case now also.